bbq/compiler/desugar.go

```diff
@@ -339,7 +410,37 @@ func (d *Desugar) VisitCompositeDeclaration(declaration *ast.CompositeDeclaratio
 	return modifiedDecl
 }
 
-func (d *Desugar) copyInheritedFunctions(compositeType *sema.CompositeType) []ast.Declaration {
+func (d *Desugar) inheritedFunctionsWithConditions(compositeType *sema.CompositeType) map[string][]*ast.FunctionDeclaration {
+	inheritedFunctions := make(map[string][]*ast.FunctionDeclaration)
+
+	for _, conformance := range compositeType.EffectiveInterfaceConformances() {
+		interfaceType := conformance.InterfaceType
+
+		// TODO: Merge this elaboration with the current elaboration (d.elaboration).
+		//   Because the inherited functions need their corresponding elaboration
+		//   for code generation.
+		elaboration, err := d.config.ElaborationResolver(interfaceType.Location)
```

Comment:
Suffers from the same problem as default-functions. Need to figure out a way to universally solve this.